### PR TITLE
XIVY-11980 implement variable search

### PR DESCRIPTION
**Pull request by ivy-lgi**
search

### CODE REVIEW

**Review by weissreto — APPROVED** [review]
👍

File: packages/new-editor/src/utils/tree/tree.ts [hunk 1]
```diff
@@ -79,3 +80,25 @@ const toTreePath = (rowId: string) => {
...
+
+export const treeGlobalFilter = <TNode extends TreeNode<TNode>>(data: Array<TNode>, path: TreePath, filterValue: string) => {
+  filterValue = filterValue.toLowerCase();
+  const nodesOnPath = getNodesOnPath(data, path);
+  for (const node of nodesOnPath) {
+    if (node?.name.toLowerCase().includes(filterValue)) {
+      return true;
+    }
+  }
+  const node = nodesOnPath[nodesOnPath.length - 1];
```
Comment (ivy-lli) [inline packages/new-editor/src/utils/tree/tree.ts:99]: if you want the last element of an array you can also do nodesOnPath.at(-1)

**Review by ivy-lli — COMMENTED** [review]
Why did you need to implement your own filter function?
Not that it is wrong, but as far as I tested it, the default filter function is looking for all columns and is case insensitive.
Or am I wrong?

**Comment by ivy-lli** [comment]
Ah, hmm...
Yes, interesting thought. Not 100% sure if this is for everyone useful, but can also not pretend otherwise.
I think we need to test this first with maybe a very large variable.yaml :)
Maybe we could later also add a setting to change this filter behavior.

Let it like that, then we can get user feedback 👍

**Review by ivy-lli — APPROVED** [review]

**Pull request merged**